adagrams/game.py

```diff
@@ -1,11 +1,151 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    #Create a set of 10 letters that matches the letter frequency distribution#
+    draw = []
+    letter_pool_copy = LETTER_POOL.copy()
+    letters_list = list(letter_pool_copy.keys())
+
+    while len(draw) < 10:       
+        #Take a random number within the len of letter pool and get the letter in that index
+        random_letter = letters_list[random.randint(0,len(letter_pool_copy)-1)]
+        
+        #Check if the letter is still availabe in the letter pool and add it to the hand
+        if letter_pool_copy[random_letter] >= 1:
+            draw.append(random_letter)
+            
+            #reduce the frequency in letter pool, for each letter added to the hand
+            letter_pool_copy[random_letter] -= 1         
+    return draw
 
 def uses_available_letters(word, letter_bank):
-    pass
+    #check if the word is an anagram of some or all of the given letters in the hand
+
+    letter_bank_copy = letter_bank.copy()
+    word = word.upper()  
+    for char in word:
+        #Take one of the letters from the hand and make unavailable in the bank copy
+        if char in letter_bank_copy:
+            letter_bank_copy.remove(char)
+        else:
+            return False
 
+    return True
+
+#-------- Score-Word Func Using Nested Lists --------#
+SCORE_CHART = [
+    [1,['A', 'E','I','O','U','L','N','R','S','T']],
+    [2,['D','G']],
+    [3,['B','C','M','P']],
+    [4,['F','H','V','W','Y']],
+    [5,['K']],
+    [8,['J','X']],
+    [10,['Q','Z']]
+]
 def score_word(word):
-    pass
+    #Returns the score of a given word as defined by the SCORE_CHART
+
+    total_points = 0
+    points = 0
+    word = word.upper()
+    for i in range(len(SCORE_CHART)):
+        for char in word:
+            if char in SCORE_CHART[i][1]: #check for char in each letter in list rows
+                points =SCORE_CHART[i][0] #If found, access the index that has the score
+                total_points += points
+            else:
+                points = 0
+
+    #Extra points if word has 7 or more characters 
+    if len(word) >= 7:
+        total_points += 8
+    
+    return total_points
```

Comment:
This is the alternative approach we were talking about in co-working:
<p>
<details>
<summary>Click this to see code.</summary>
<pre>
SCORE_DICT = {
    "A": 1,
    "E": 1,
    "I": 1,
    "O": 1,
    "U": 1,
    "L": 1,
    "N": 1,
    "R": 1,
    "S": 1,
    "T": 1,
    "D": 2,
    "G": 2,
    "B": 3,
    "C": 3,
    "M": 3,
    "P": 3,
    "F": 4,
    "H": 4,
    "V": 4,
    "W": 4,
    "Y": 4,
    "K": 5,
    "J": 8,
    "X": 8,
    "Q": 10,
    "Z": 10
}

def score_word(word):
    score += 0

    casefold_word = word.casefold()

    for letter in casefold_word:
        score += SCORE_DICT.get(letter)
    
    if len(casefold_word) > 6:
        score += 8
    
    return score
</pre>